File: app/src/main/java/com/adsamcik/signalcollector/Receivers/BootReceiver.java
package com.adsamcik.signalcollector.Receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.adsamcik.signalcollector.Play.PlayController;

public class BootReceiver extends BroadcastReceiver {

	@Override
	public void onReceive(Context context, Intent intent) {
		if (intent.getAction().equals(Intent.ACTION_BOOT_COMPLETED))
			PlayController.initializeActivityClient(context);
	}
}

File: app/src/main/java/com/adsamcik/signalcollector/receivers/BootReceiver.kt
package com.adsamcik.signalcollector.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.adsamcik.signalcollector.activities.LaunchActivity
import com.adsamcik.signalcollector.services.ActivityService
import com.adsamcik.signalcollector.services.ActivityWakerService

class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            ActivityWakerService.poke(context)
            ActivityService.requestAutoTracking(context, LaunchActivity::class.java)
        }
    }
}
